### PR TITLE
feat: add bun support

### PR DESCRIPTION
**Pull request by wcastand**
to support : #87 

- added bun to templates (need to verify --cwd does what i expect)
- added bun detection in the package manager
- add to the github action bun bin (`- uses: oven-sh/setup-bun@v2`)

left to do as far as i know:
- see if gluegun can support bun easily or not
- check what kind of tests i need to run to check it works as intended

### CODE REVIEW

**Comment by wcastand** [comment]
currently block since i realize `gluegun` doesn't support `bun` as a package manager.
https://github.com/infinitered/gluegun/issues/794

**Comment by maciekstosio** [comment]
> currently block since i realize gluegun doesn't support bun as a package manager.
Yeah, that's why we didn't include bun and pnpm in the alpha version. If you're interested in contributing I think create-expo-stack handles all of them and we could learn from it: https://github.com/roninoss/create-expo-stack/blob/main/cli/src/utilities/getPackageManager.ts

**Comment by wcastand** [comment]
from what i understood of their codebase (quick look might be more to that)
they basically don't use gluegun packageManager, what block us is we use `force` option of gluegun packageManager from toolbox and from what i can see, they don't use the packageManager from the toolbox and handle it on their own or differently(didn't find an equivalent yet from my quick look)

File: src/templates/detox/test-detox-android.ejf [hunk 1]
```diff
@@ -39,6 +41,11 @@ jobs:
       - name: 🏗 Checkout repository
         uses: actions/checkout@v4
 
+      <%= if(props.packageManager === "bun") { %>
+      - name: Install Bun
```
Comment (szarbartosz) [inline src/templates/detox/test-detox-android.ejf:45]:
```suggestion
      - name: 🥟 Install Bun
```

File: src/templates/detox/test-detox-ios.ejf [hunk 2]
```diff
@@ -37,6 +39,11 @@ jobs:
       - name: 🏗 Checkout repository
         uses: actions/checkout@v4
 
+      <%= if(props.packageManager === "bun") { %>
+      - name: Install Bun
```
Comment (szarbartosz) [inline src/templates/detox/test-detox-ios.ejf:43]:
```suggestion
      - name: 🥟 Install Bun
```

File: src/templates/eas/eas-update.ejf [hunk 3]
```diff
@@ -31,6 +32,11 @@ jobs:
       - name: 🏗 Setup repo
         uses: actions/checkout@v4
 
+      <%= if(props.packageManager === "bun") { %>
+      - name: Install Bun
```
Comment (szarbartosz) [inline src/templates/eas/eas-update.ejf:36]:
```suggestion
      - name: 🥟 Install Bun
```

File: src/templates/jest/jest.ejf [hunk 4]
```diff
@@ -27,6 +29,11 @@ jobs:
          - name: 🏗 Setup repo
            uses: actions/checkout@v4
 
+        <%= if(props.packageManager === "bun") { %>
+         - name: Install Bun
```
Comment (szarbartosz) [inline src/templates/jest/jest.ejf:33]:
```suggestion
      - name: 🥟 Install Bun
```

File: src/templates/lint/lint.ejf [hunk 5]
```diff
@@ -27,6 +29,11 @@ jobs:
          - name: 🏗 Setup repo
            uses: actions/checkout@v4
 
+        <%= if(props.packageManager === "bun") { %>
+         - name: Install Bun
```
Comment (szarbartosz) [inline src/templates/lint/lint.ejf:33]:
```suggestion
         - name: 🥟 Install Bun
```

File: src/templates/prettier/prettier.ejf [hunk 6]
```diff
@@ -27,6 +29,11 @@ jobs:
          - name: 🏗 Setup repo
            uses: actions/checkout@v4
 
+        <%= if(props.packageManager === "bun") { %>
+         - name: Install Bun
```
Comment (szarbartosz) [inline src/templates/prettier/prettier.ejf:33]:
```suggestion
         - name: 🥟 Install Bun
```

**Comment by maciekstosio** [comment]
Solved by #133

**Pull request closed without merging**